### PR TITLE
MLIBZ-2666: "+" sign is removed from query.

### PR DESCRIPTION
**Pull request by romanogolikhin**
#### Description
There is not any URL query encoding in Kinvey HTTP requests. As a result the sign "+" is being lost, when a request reaches a server side.

#### Changes
Changes in the "NetworkFactory" class. Adding URL encoding for query in HTTP requests.

#### Tests
TestDeltaSetWithQueryPullReturnCorrectNumberOfUpdates()
TestDeleteByQueryStringValueWithPlusSymbolEqualExpressionAsync()
TestGetCountAsyncWithQuery()
TestNetworkStoreFindByMongoQueryWithPlusSymbol()

### CODE REVIEW

File: Kinvey.Core/Core/NetworkFactory.cs [hunk 1]
```diff
@@ -286,7 +286,8 @@ public NetworkRequest<T> BuildDeltaSetRequest<T>(string collectionName, string l
             {
                 restPathParams += string.IsNullOrEmpty(restPathParams) ? "?" : "&";
                 restPathParams += "query={query}";
-                urlParameters.Add("query", query);
+                var encodedQuery = System.Net.WebUtility.UrlEncode(query);
```
Comment (heyzooi) [inline Kinvey.Core/Core/NetworkFactory.cs]: @romanogolikhin I'm not entirely sure if we should use `System.Net.WebUtility.UrlEncode()` or `Uri.EscapeDataString()`. Reading the docs and some links in the internet it seems very confusing the difference between those 2. I'm wondering if you have an opinion about it.

Comment (heyzooi) [reply Kinvey.Core/Core/NetworkFactory.cs]: not 100% sure yet, but it seems `System.Uri.EscapeDataString()` is what we need in this case

https://stackoverflow.com/questions/602642/server-urlencode-vs-httputility-urlencode

Comment (romanogolikhin) [reply Kinvey.Core/Core/NetworkFactory.cs]: @heyzooi  I have read what you had recommended me. I agree with you.
The "System.Uri.EscapeDataString()" is more preferable for us.
I have committed changes.

**Comment by heyzooi** [comment]
@romanogolikhin before you merge please check why CI is failing

**Pull request merged**